crates/wysiwyg/src/dom/nodes/mention_node.rs

```diff
@@ -129,22 +129,34 @@ impl<S: UnicodeString> MentionNode<S> {
         let cur_pos = formatter.len();
         match self.kind() {
             MentionNodeKind::MatrixUrl { display_text, url } => {
-                let mut attributes = self.attributes.clone();
-                attributes.push(("href".into(), url.clone()));
-
-                if !as_message {
-                    attributes.push(("contenteditable".into(), "false".into()))
-                    // TODO: data-mention-type = "user" | "room"
-                }
+                // if formatting as a message, only include the href attribute
+                let attributes = if as_message {
+                    vec![("href".into(), url.clone())]
+                } else {
+                    let mut attributes_for_composer = self.attributes.clone();
+                    attributes_for_composer.push(("href".into(), url.clone()));
+                    attributes_for_composer
+                        .push(("contenteditable".into(), "false".into()));
+                    attributes_for_composer
+                };
 
                 self.fmt_tag_open(tag, formatter, &Some(attributes));
-
                 formatter.push(display_text.clone());
-
                 self.fmt_tag_close(tag, formatter);
             }
             MentionNodeKind::AtRoom => {
-                formatter.push(self.display_text());
+                // if formatting as a message, simply use the display text (@room)
+                if as_message {
+                    formatter.push(self.display_text())
+                } else {
+                    let mut attributes = self.attributes.clone();
+                    attributes.push(("href".into(), "#".into())); // designates a placeholder link in html
+                    attributes.push(("contenteditable".into(), "false".into()));
+
+                    self.fmt_tag_open(tag, formatter, &Some(attributes));
```

Comment:
Should this tag be `<span>` (or `<mention>`), rather than `<a>`?

Reply:
For the time being keeping everything as `<a>` - [there's a separate issue for moving to a custom html tag type here](https://github.com/matrix-org/matrix-rich-text-editor/issues/714) to try and keep these PRs small and digestible